### PR TITLE
OpenApiMembersManager

### PR DESCRIPTION
**Pull request by xpavlic**
- MembersManager methods used by GUI implemented into openApi

### CODE REVIEW

**Comment by zlamalp** [comment]
Please re-base this pull-request, thank you.

File: perun-openapi/openapi.yml [hunk 1]
```diff
@@ -8140,6 +8206,241 @@ paths:
...
+        - MembersManager
+      operationId: FindCompleteRichMembersByAttributes
+      summary: Return list of richMembers from perun by the searchString with attributes specific for list of attrsNames
+        and who have only status which is contain in list of statuses. If attrsNames is empty or null
+        return all attributes for specific richMembers. If listOfStatuses is empty or null, return all possible statuses.
+      parameters:
+        - $ref: '#/components/parameters/listOfAttributesNames'
+        - $ref: '#/components/parameters/searchString'
+        - $ref: '#/components/parameters/lookingInParentGroup'
+        - name: "allowedStatuses[]"
```
Comment (balcirakpeter) [inline perun-openapi/openapi.yml:8243]: I think this attribute should be optional. Therefore, you can reference it.

File: perun-openapi/openapi.yml [hunk 2]
```diff
@@ -8140,6 +8206,241 @@ paths:
...
+      tags:
+        - MembersManager
+      operationId: FindCompleteRichMembersByAttributes
+      summary: Return list of richMembers from perun by the searchString with attributes specific for list of attrsNames
+        and who have only status which is contain in list of statuses. If attrsNames is empty or null
+        return all attributes for specific richMembers. If listOfStatuses is empty or null, return all possible statuses.
+      parameters:
+        - $ref: '#/components/parameters/listOfAttributesNames'
+        - $ref: '#/components/parameters/searchString'
+        - $ref: '#/components/parameters/lookingInParentGroup'
```
Comment (balcirakpeter) [inline perun-openapi/openapi.yml:8242]: I think when there is no groupId, this parameter does not make sense

File: perun-openapi/openapi.yml [hunk 3]
```diff
@@ -8140,6 +8206,241 @@ paths:
...
+    get:
+      tags:
+        - MembersManager
+      operationId: getCompleteRichMembersForVo
+      summary: Get all RichMembers with attributes specific for list of attrsNames from the vo and have only status
+        which is contain in list of statuses. If attrsNames is empty or null return all attributes for specific richMembers.
+        If listOfStatuses is empty or null, return all possible statuses.
+      parameters:
+        - $ref: '#/components/parameters/voId'
+        - $ref: '#/components/parameters/listOfAttributesNames'
```
Comment (balcirakpeter) [inline perun-openapi/openapi.yml:8361]: This referenced attribute is set as required but I think for this scenario it should be optional.

File: perun-openapi/openapi.yml [hunk 4]
```diff
@@ -8140,6 +8206,241 @@ paths:
...
+        - MembersManager
+      operationId: getCompleteRichMembersForGroup
+      summary: Get all RichMembers with attributes specific for list of attrsNames from the group and have only status
+        which is contain in list of statuses. If attrsNames is empty or null return all attributes for specific richMembers.
+        If listOfStatuses is empty or null, return all possible statuses. If lookingInParentGroup is true,
+        get all these richMembers only for parentGroup of this group. If this group is top level group,
+        so get richMembers from members group.
+      parameters:
+        - $ref: '#/components/parameters/groupId'
+        - $ref: '#/components/parameters/listOfAttributesNames'
```
Comment (balcirakpeter) [inline perun-openapi/openapi.yml:8381]: Same here.

**Review by balcirakpeter — CHANGES_REQUESTED** [review]

File: perun-openapi/openapi.yml [hunk 5]
```diff
@@ -8122,11 +8153,46 @@ paths:
         default:
           $ref: '#/components/responses/ExceptionResponse'
 
+  /json/membersManager/deleteMembers:
```
Comment (Vojtech-Sassmann) [inline perun-openapi/openapi.yml:8156]:
```suggestion
  /urlinjsonout/membersManager/deleteMembers:
```

File: perun-openapi/openapi.yml (hunk 3 above)
Comment (Vojtech-Sassmann) [reply perun-openapi/openapi.yml:8361]: You can use `attrsNames` which is already declared as a parameter but you have to fix its name. I have noticed it is wrong `attrsNames` -> `attrsNames[]`

File: perun-openapi/openapi.yml [hunk 6]
```diff
@@ -8140,6 +8206,241 @@ paths:
...
+  /urlinjsonout/membersManager/setStatus:
+    post:
+      tags:
+        - MembersManager
+      operationId: setStatus
+      summary: Set membership status of a member.
+      parameters:
+        - $ref: '#/components/parameters/memberId'
+        - name: status
+          description: status (VALID | INVALID | SUSPENDED | EXPIRED | DISABLED)
```
Comment (Vojtech-Sassmann) [inline perun-openapi/openapi.yml:8426]:
```suggestion
          description: status (VALID | INVALID | EXPIRED | DISABLED)
```

File: perun-openapi/openapi.yml [hunk 7]
```diff
@@ -8140,6 +8206,241 @@ paths:
...
+          schema:
+            type: string
+          in: query
+          required: true
+        - name: message
+          description: "reason for suspension"
+          schema:
+            type: string
+          in: query
+          required: false
```
Comment (Vojtech-Sassmann) [inline perun-openapi/openapi.yml:8436]: This is no longer possible, please, remove it.

**Review by Vojtech-Sassmann — CHANGES_REQUESTED** [review]

**Review by Vojtech-Sassmann — APPROVED** [review]

**Review by balcirakpeter — APPROVED** [review]

**Pull request merged**